s390x precision seems to be a little bit off in some operations

On s390x the precision of some operation makes TestExecFile fail.

```
# go test -run "^TestExecFile$" go.starlark.net/starlark
--- FAIL: TestExecFile (0.35s)
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:186:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: 0.7853981633974484 != 0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:187:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: -0.7853981633974484 != -0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:188:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: 0.7853981633974484 != 0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:189:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: -0.7853981633974484 != -0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:195:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: 0.7853981633974484 != 0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:196:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: -0.7853981633974484 != -0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:202:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: 0.7853981633974484 != 0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:203:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: -0.7853981633974484 != -0.7853981633974483
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:310:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: 0.5493061443340549 != 0.5493061443340548
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:311:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: -0.5493061443340549 != -0.5493061443340548
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:348:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: -0.761594155955765 != -0.7615941559557649
    starlarktest.go:118: Traceback (most recent call last):
          testdata/math.star:349:10: in <toplevel>
          assert.star:14:14: in _eq
        Error: -0.761594155955765 != -0.7615941559557649
FAIL
FAIL    go.starlark.net/starlark        0.376s
FAIL
```

I tried with several versions of Go (from 1.16 up to 1.20.4) and the results are always the same.

--- a/starlarktest/assert.star
+++ b/starlarktest/assert.star
@@ -6,12 +6,20 @@
 # matches(str, pattern): report whether str matches regular expression pattern.
 # module(**kwargs): a constructor for a module.
 # _freeze(x): freeze the value x and everything reachable from it.
+# _floateq(x, y): reports floating point equality (within 1 ULP).
 #
 # Clients may use these functions to define their own testing abstractions.
 
+_num = ("float", "int")
+
 def _eq(x, y):
     if x != y:
-        error("%r != %r" % (x, y))
+	if (type(x) == "float" and type(y) in _num or
+	    type(y) == "float" and type(x) in _num):
+	    if not _floateq(float(x), float(y)):
+		error("floats: %r != %r (delta > 1 ulp)" % (x, y))
+	else:
+            error("%r != %r" % (x, y))
 
 def _ne(x, y):
     if x == y:

--- a/starlarktest/starlarktest.go
+++ b/starlarktest/starlarktest.go
@@ -15,6 +15,7 @@ package starlarktest // import "go.starlark.net/starlarktest"
 import (
 	_ "embed"
 	"fmt"
+	"math"
 	"os"
 	"path/filepath"
 	"regexp"
@@ -63,11 +64,12 @@ var (
 func LoadAssertModule() (starlark.StringDict, error) {
 	once.Do(func() {
 		predeclared := starlark.StringDict{
-			"error":   starlark.NewBuiltin("error", error_),
-			"catch":   starlark.NewBuiltin("catch", catch),
-			"matches": starlark.NewBuiltin("matches", matches),
-			"module":  starlark.NewBuiltin("module", starlarkstruct.MakeModule),
-			"_freeze": starlark.NewBuiltin("freeze", freeze),
+			"error":    starlark.NewBuiltin("error", error_),
+			"catch":    starlark.NewBuiltin("catch", catch),
+			"matches":  starlark.NewBuiltin("matches", matches),
+			"module":   starlark.NewBuiltin("module", starlarkstruct.MakeModule),
+			"_freeze":  starlark.NewBuiltin("freeze", freeze),
+			"_floateq": starlark.NewBuiltin("floateq", floateq),
 		}
 		thread := new(starlark.Thread)
 		assert, assertErr = starlark.ExecFile(thread, "assert.star", assertFileSrc, predeclared)
@@ -129,6 +131,29 @@ func freeze(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, k
 	}
 	args[0].Freeze()
 	return args[0], nil
+}
+
+// floateq(x, y) reports whether two floats are within 1 ULP of each other.
+func floateq(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var xf, yf starlark.Float
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 2, &xf, &yf); err != nil {
+		return nil, err
+	}
+
+	res := false
+	switch {
+	case xf == yf:
+		res = true
+	case math.IsNaN(float64(xf)):
+		res = math.IsNaN(float64(yf))
+	case math.IsNaN(float64(yf)):
+		// false (non-NaN = Nan)
+	default:
+		x := math.Float64bits(float64(xf))
+		y := math.Float64bits(float64(yf))
+		res = x == y+1 || y == x+1
+	}
+	return starlark.Bool(res), nil
 }
 
 // DataFile returns the effective filename of the specified

